packages/dialect-react-ui/components/Notifications/screens/NewSettings/Email.tsx

```diff
@@ -0,0 +1,203 @@
+import { Toggle } from '../../../common';
+import { useTheme } from '../../../common/providers/DialectThemeProvider';
+import clsx from 'clsx';
+import { AddressType, useAddresses } from '@dialectlabs/react-sdk';
+import { useEffect, useState } from 'react';
+import { P } from '../../../common/preflighted';
+import { VerificationInput } from './VerificationInput';
+import { RightAdornment } from './RightAdorment';
+import OutlinedInput from '../../../common/primitives/OutlinedInput';
+
+const addressType = AddressType.Email;
+
+const Email = () => {
+  const { textStyles } = useTheme();
+
+  const {
+    addresses: { [addressType]: emailAddress },
+    create: createAddress,
+    delete: deleteAddress,
+    update: updateAddress,
+
+    toggle: toggleAddress,
+
+    isCreatingAddress,
+    isUpdatingAddress,
+    isDeletingAddress,
+    isSendingCode,
+    isVerifyingCode,
+
+    errorFetching: errorFetchingAddresses,
+  } = useAddresses();
+
+  const [email, setEmail] = useState(emailAddress?.value ?? '');
+  const [isEmailEditing, setEmailEditing] = useState(!emailAddress?.enabled);
+  const [isEnabled, setIsEnabled] = useState(Boolean(emailAddress?.enabled));
+
+  const [error, setError] = useState<Error | null>(null);
+
+  const isEmailSaved = Boolean(emailAddress);
+  const isChanging = emailAddress && isEmailEditing;
+  const isVerified = emailAddress?.verified;
+
+  const isLoading =
+    isCreatingAddress ||
+    isDeletingAddress ||
+    isUpdatingAddress ||
+    isVerifyingCode ||
+    isSendingCode;
+
+  const currentError = error || errorFetchingAddresses;
+
+  // // FIXME: replace with key change
+  useEffect(() => {
+    // Update state if addresses updated
+    setEmailEditing(!isEmailSaved && !emailAddress?.enabled);
+    setEmail(emailAddress?.value || '');
+  }, [isEmailSaved, emailAddress?.enabled, emailAddress?.value]);
+
+  const updateEmail = async () => {
+    if (error) {
+      return;
+    }
+    try {
+      await updateAddress({ addressType, value: email });
+      setError(null);
+    } catch (e) {
+      setError(e as Error);
+    }
+    setEmailEditing(false);
+  };
+
+  const saveEmail = async () => {
+    if (error) {
+      return;
+    }
+    try {
+      await createAddress({ addressType, value: email });
+      setError(null);
+    } catch (e) {
+      setError(e as Error);
+    }
+  };
+
+  const deleteEmail = async () => {
+    try {
+      await deleteAddress({ addressType });
```

Comment:
and one more time about deletion, guess we shouldn't delete address, but only dapp address? @tsmbl do you have insights on this?

Reply:
technically it's ok to delete it, I think, but consequences MUST be explicitly communicated to user